src/classes/MergeService.cls

```diff
@@ -49,11 +61,17 @@ global with sharing class MergeService {
         // Return the fully constructed SOQL statement
         return String.join(new List<String> {
             selectClause,
-            'FROM Account',
+            'FROM '+sobjectName,
```

Comment:
I've taken to being more explicit in code to using `this` to denote whether the variable is a local, method-scoped variable or a _field_ (see [Java's definition of a "field"][1]) in the class. So in this case, it's better to use `this.sobjectName` instead of `sobjectName` even though the latter does still resolve correctly.

[1]: https://docs.oracle.com/javase/tutorial/reflect/member/fieldValues.html